resolwe_bio/processes/reads_processing/bamclipper.py

```diff
@@ -46,38 +52,51 @@ class Output:
     def run(self, inputs, outputs):
         """Run the analysis."""
         bam_build = inputs.alignment.build
-        bedpe_build = inputs.bedpe.build
         bam_species = inputs.alignment.species
-        bedpe_species = inputs.bedpe.species
-
-        if bam_build != bedpe_build:
-            self.error(
-                f'Builds of the genome {bam_build} and annotation '
-                f'{bedpe_build} do not match. Please provide genome and '
-                f'annotation with the same build.'
-            )
-
-        if bam_species != bedpe_species:
-            self.error(
-                f'Species of BAM ({bam_species}) and BEDPE ({bedpe_species}) '
-                'files do not match.'
-            )
-
-        # Output of bamclipper.sh is a file that is appended a primerclipped just before file
-        # extension, e.g. "file.bam" now becomes "file.primerclipped.bam".
-        bc_inputs = [
-            '-b',
-            f'{inputs.alignment.bam.path}',
-            '-p',
-            f'{inputs.bedpe.bedpe.path}'
-        ]
-        Cmd['bamclipper.sh'](bc_inputs)
 
         name = os.path.splitext(os.path.basename(inputs.alignment.bam.path))[0]
-        bam = f'{name}.primerclipped.bam'
-        bai = f'{name}.primerclipped.bam.bai'
 
-        self.progress(0.5)
+        # If so specified, skip bamclipper step. Prepare outputs to match those of as if bamclipping proceeded.
+        if inputs.skip:
+            bam = f'{name}.bamclipper_skipped.bam'
+            bai = f'{bam}.bai'
+            bigwig = f'{bam[:-4]}.bw'
+
+            copy2(inputs.alignment.bam.path, bam)
```

Comment:
What about the copy of `bai` and `bigwig` files? I don't see where those get created in the case of `skip = True`

Reply:
Oh, I see they are created either way.